File: algorithms/2_medium/M/01911_maximum_alternating_subsequence_sum/greedy.cpp
class Solution {
public:
    static long long maxAlternatingSum(const vector<int>& nums) {
        int prev_num = 0;
        long long curr_sum = 0;

        for (const int num : nums) {
            if (num > prev_num) {
                curr_sum += num - prev_num;
            }
            prev_num = num;
        }

        return curr_sum;
    }
};

File: algorithms/2_medium/M/01911_maximum_alternating_subsequence_sum/greedy.rs
impl Solution {
    pub fn max_alternating_sum(nums: Vec<i32>) -> i64 {
        let mut prev_num: i32 = 0;
        let mut curr_sum: i64 = 0;

        for num in nums {
            debug_assert!(num > 0);
            curr_sum += (num as i64 - prev_num as i64).max(0);
            prev_num = num;
        }

        curr_sum
    }
}
